File: src/page/JTG.java
package page;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.function.Consumer;

import javax.swing.JToggleButton;

import common.util.lang.LocaleCenter.Binder;

public class JTG extends JToggleButton implements LocComp {

	private static final long serialVersionUID = 1L;

	private final LocSubComp lsc;

	public JTG() {
		lsc = new LocSubComp(this);
	}

	public JTG(Binder binder) {
		this();
		lsc.init(binder);
	}

	public JTG(int i, String str) {
		this();
		lsc.init(i, str);
	}

	public JTG(String str) {
		this(-1, str);
	}

	@Override
	public LocSubComp getLSC() {
		return lsc;
	}

	public void setLnr(Consumer<ActionEvent> c) {
		addActionListener(new ActionListener() {

			@Override
			public void actionPerformed(ActionEvent e) {
				c.accept(e);
			}

		});
	}

}

File: src/main/java/page/JTG.kt
package page

import common.util.lang.LocaleCenter
import java.awt.event.ActionEvent
import java.util.function.Consumer
import javax.swing.JToggleButton

class JTG() : JToggleButton(), LocComp {
    private val lsc: LocSubComp = LocSubComp(this)

    constructor(binder: LocaleCenter.Binder?) : this() {
        lsc.init(binder)
    }

    constructor(i: Int, str: String?) : this() {
        lsc.init(i, str)
    }

    constructor(str: String?) : this(-1, str)

    override fun getLSC(): LocSubComp {
        return lsc
    }

    fun setLnr(c: Consumer<ActionEvent?>) {
        addActionListener { e -> c.accept(e) }
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}
